CS3230Project/View/Checkup.cs

```diff
@@ -0,0 +1,143 @@
+{
+    /// <summary>
+    ///   The Checkup form
+    /// </summary>
+    public partial class Checkup : Form
+    {
+        private int appointmentID;
+        private readonly string invalidInputErrorMessage = "Invalid Values for the checkup details";
+        private readonly string invalidInputErrorHeader = "Unable to add new checkup";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Checkup" /> class.
+        /// </summary>
+        /// <param name="appointmentID">The appointment identifier.</param>
+        public Checkup(int appointmentID)
+        {
+            InitializeComponent();
+            this.submitChangesFooter1.SubmitButtonEventHandler += this.submitChangesFooter1OnSubmitButtonEventHandler;
+            this.submitChangesFooter1.BackButtonEventHandler += this.SubmitChangesFooter1OnBackButtonEventHandler;
+            this.header1.LogoutEventHandler += this.Header1OnLogoutEventHandler;
+            this.appointmentID = appointmentID;
+            this.populateNurseComboBox();
+        }
+
+        private void populateNurseComboBox()
+        {
+            foreach (var nurse in NurseManager.GetNurses())
+            {
+                var nurseInfo = nurse.FirstName + " " + nurse.LastName + " ID: " + nurse.NurseId;
+                this.nurseComboBox.Items.Add(nurseInfo);
+            }
+        }
+
+        private void Header1OnLogoutEventHandler(object sender, EventArgs e)
+        {
+            SwitchForms.SwitchToLogin(this);
+        }
+
+        private void SubmitChangesFooter1OnBackButtonEventHandler(object sender, EventArgs e)
+        {
+            SwitchForms.SwitchBackToHome(this);
+        }
+
+        private void submitChangesFooter1OnSubmitButtonEventHandler(object sender, EventArgs e)
+        {
+            try
+            {
+                this.verifyAll();
+
+                var nurseID = this.getNurseID((string)this.nurseComboBox.Text);
+                Visit visitToAdd = new Visit(this.appointmentID, nurseID, Convert.ToDouble(this.bodyTemperatureTextBox.Text), 
+                    Convert.ToInt16(this.pulseTextBox.Text), Convert.ToDouble(this.heightTextBox.Text),
+                    Convert.ToDouble(this.weightTextBox.Text), this.symptomsTextBox.Text, 
+                    Convert.ToInt16(this.systolicBloodPressureTextBox.Text), 
+                    Convert.ToInt16(this.diastolicBloodPressureTextBox.Text));
+                CheckupManagerViewModel.AddVisit(visitToAdd);
+                Form searchPatientForm = new SearchPatient();
+                SwitchForms.Switch(this, searchPatientForm);
+            }
+            catch (Exception)
+            {
+                MessageBox.Show(this.invalidInputErrorMessage, this.invalidInputErrorHeader);
+            }
+        }
+
+        private int getNurseID(string nurseInfo)
+        {
+            string[] nurseInfoSplit = nurseInfo.Split(' ');
+
+            if (nurseInfoSplit.Length > 2)
```

Comment:
I would be nice to extract this into a field, I don't know what 2 is supposed to mean,.

Reply:
I will fix it.